### ISSUE TITLE
Question about the  peak definition.

### ISSUE THREAD

**Issue by fatyang799** [guanjue/S3V2_IDEAS_ESMP#10]
Hi Guanjue:

Since I read the source code for S3V2_IDEAS_ESMP 1 year ago, I've been borrowing from your code in defining peak (get_common_pk.R). However, I suddenly realize that there seems to be a problem with this code. 

```R
### get z
get_z = function(x){
	x_notop = x[x<=quantile(x, 0.99)]
    xz = (x - mean(x_notop)) / sd(x_notop)
    return(xz)
}

### get fdr
get_fdr = function(x){
    z = get_z(x)
    zp = pnorm(-abs(z))
    zpfdr = p.adjust(zp)
    return(zpfdr)
}
```

The function `get_z` is the `zscale` normalization, which you use only value more than Q99 data. I agree with this idea. But for the function `get_fdr`, why do you write `zp = pnorm(-abs(z))`? In this way, you define not only the signal with the highest value but also the signal with the lowest value as the peaks.

To make it easier to explain, I use a set of dummy data for this explanation.

```bash
set.seed(2023)
a <- rnorm(1000)

### get fdr
get_fdr = function(z){
  zp = pnorm(-abs(z))
  zpfdr = p.adjust(zp)
  return(zpfdr)
}
res <- data.frame(a=a, 
                  fdr=get_fdr(a))

plot(res$a, res$fdr)
```

![test](https://github.com/guanjue/S3V2_IDEAS_ESMP/assets/52167840/6e103413-7629-4f49-b185-5409b4191657)

you can see a value with the lowest value can still be a peak when get fdr value with `zp = pnorm(-abs(z))`. So I'm wondering if I need to change the code to the following:

```R
### get z
get_z = function(x){
	x_notop = x[x<=quantile(x, 0.99)]
    xz = (x - mean(x_notop)) / sd(x_notop)
    return(xz)
}

### get fdr
get_fdr = function(x){
    z = get_z(x)
    zp = pnorm(z, lower.tail = F)
    zpfdr = p.adjust(zp)
    return(zpfdr)
}
```

I would greatly appreciate if you could give me some suggestions.

**Comment by guanjue** [comment]
You're absolutely right, and I appreciate your keen observation. This was indeed an error in the script. We have now corrected it. 

Regarding the implications of the previous analysis: Fortunately, regions with the lowest signals are likely to be randomly distributed across large number of background regions in different cell types. Consequently, the likelihood of mistakenly identifying these regions as "common peak regions" across all cell types is minimal. This is why, we did not find this error when we briefly checked our common peak regions along the genome. 

Thank you for bringing this to our attention.
